realm/realm-library/src/androidTest/java/io/realm/internal/CollectionTests.java

```diff
@@ -396,15 +398,16 @@ public void onChange(Collection collection1) {
                         assertEquals(collection1.size(), 4);
                         break;
                     case 1:
+                        assertFalse(transactionCommitted.get());
                         assertEquals(collection1.size(), 5);
                         sharedRealm.close();
-                        looperThread.testComplete();
                         break;
                 }
             }
         });
         addRow(sharedRealm);
-        assertEquals(collection.size(), 5);
+        transactionCommitted.set(true);
+        looperThread.testComplete();
```

Comment:
See the diff of the comments at the beginning of the test, the original test is to test the `commitTransaction` will trigger the listener in the next event loop. But now, the behavior to be tested is changed.